sys/src/libacpi/harvey.c

```diff
@@ -436,22 +436,65 @@ acpihandler(void *_, void *arg)
 }
 #endif
 
+static struct handler ihandler;
+static Rendez irendez;
+static QLock irlock;
+static int intrpid;
+
+/*
+ * The acpica service routines can't run in the note handler
+ * because of a floating point exception. Fixable?
```

Comment:
we should not assume it is. what you are doing with the rendezvous is fine BUT: you could use the plan 9 threads and channels model, and have  a process sleeping on a chan. The acpihandler would write useful messages down the chan, which will wake up the thread.